### PR TITLE
Convert renderer to unique_ptr

### PR DESCRIPTION
**Pull request by ShFil119**

### CODE REVIEW

**Comment by madebr** [comment]
I believe the `std::shared_ptr` abstraction was made as to allow different renderers.
Maybe use a `std::variant`?

**Comment by danhedron** [comment]
The polymorphism here was to allow different render implementations, despite the OpenGL that leaked into other places.

File: rwgame/states/PauseState.cpp [hunk 1]
```diff
@@ -33,7 +33,7 @@ void PauseState::tick(float dt) {
 void PauseState::draw(GameRenderer* r) {
```
Comment (madebr) [inline rwgame/states/PauseState.cpp]: Can this become a reference?

File: rwgame/states/LoadingState.cpp [hunk 2]
```diff
@@ -32,7 +32,7 @@ void LoadingState::draw(GameRenderer* r) {
     // Display some manner of loading screen.
     TextRenderer::TextInfo ti;
     ti.text = kLoadingString;
-    auto size = r->getRenderer()->getViewport();
+    auto size = r->getRenderer().getViewport();
```
Comment (madebr) [inline rwgame/states/LoadingState.cpp]: Can r become a reference as well?

File: rwgame/HUDDrawer.cpp [hunk 3]
```diff
@@ -13,7 +13,7 @@
 void HUDDrawer::drawScriptTimer(GameWorld* world, GameRenderer* render) {
```
Comment (madebr) [inline rwgame/HUDDrawer.cpp]: `render` reference as well?

**Comment by madebr** [comment]
grep gave me this:
```
$ grep -rn "GameRenderer *\\*"
rwviewer/ViewerWindow.hpp:48:    void gameLoaded(GameWorld*, GameRenderer*);
rwviewer/ViewerWidget.hpp:64:    void gameLoaded(GameWorld* world, GameRenderer* renderer);
rwviewer/ViewerWidget.hpp:83:    GameRenderer* _renderer = nullptr;
rwviewer/views/ViewerInterface.hpp:26:    void gameLoaded(GameWorld*, GameRenderer*);
rwviewer/views/ViewerInterface.hpp:29:    void showData(GameWorld* world, GameRenderer* renderer) {
rwviewer/ViewerWidget.cpp:332:void ViewerWidget::gameLoaded(GameWorld *world, GameRenderer *renderer) {
```

**Comment by ShFil119** [comment]
@madebr, I won't touch viewer in this pull, because renderer is initialized in function `gameLoaded` and even if there's possibility to initialize that variable in ctor, it would require "some" more work.

**Pull request merged**